### PR TITLE
Outlook Mail: map Outlook IDs to auto-incrementing numerical IDs

### PR DESCRIPTION
**Pull request by g-linville**
gpt-4o was getting confused by the long base64 IDs that Outlook uses. This change swaps all of those for auto-incrementing numerical IDs, which are much easier for the LLM to handle. We store the mapping in a SQLite DB stored in the workspace directory, and we create new entries on the fly as needed. The mapping is indexed on both fields so we can do reverse lookups performantly as well, to avoid unnecessarily creating the same entry in the mapping multiple times.

For now, this is implemented only in Mail, but I made the mapping stuff its own package so that we can easily include it in Calendar. Based on my testing, this seems to work very well, but I want a couple other people to try it before I bother to do it in the Calendar code.

### CODE REVIEW

File: apis/outlook/common/go.mod
Comment (g-linville) [inline apis/outlook/common/go.mod:6]: This is the pure Go (no CGo) implementation of the SQLite driver for GORM.

File: apis/outlook/mail/code/go.mod
Comment (g-linville) [inline apis/outlook/mail/code/go.mod:5]: This is pointing at my fork for now. Unfortunately I can't really change this until after I merge this PR. Should I open up a second PR to introduce the `common` package by itself first?

Comment (thedadams) [reply apis/outlook/mail/code/go.mod:5]: I think it's fine to merge and immediately create another PR.

File: apis/outlook/mail/code/pkg/commands/searchMessages.go
Comment (thedadams) [inline apis/outlook/mail/code/pkg/commands/searchMessages.go:34]: nit: don't need the `else` block anymore.

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (thedadams) [inline apis/outlook/mail/code/pkg/printers/mail.go:86]: I would appreciate a more descriptive name for this, or a comment explaining exactly what is being fixed.

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (thedadams) [inline apis/outlook/mail/code/pkg/printers/mail.go:109]: Same comment here.

**Review by thedadams — CHANGES_REQUESTED** [review]

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (StrongMonkey) [inline apis/outlook/mail/code/pkg/printers/mail.go:96]: Hmm I think `folder.SetId` should just work?

File: apis/outlook/mail/code/go.mod
Comment (StrongMonkey) [reply apis/outlook/mail/code/go.mod:5]: any reason this common package has to be separate go module?

**Review by StrongMonkey — APPROVED** [review]
LGTM once donnie's comment has been addressed.

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (g-linville) [reply apis/outlook/mail/code/pkg/printers/mail.go:96]: No clue why, but it wasn't working, which is why I had to do it this way.

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (g-linville) [reply apis/outlook/mail/code/pkg/printers/mail.go:86]: Yeah I should have named these better lol

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (g-linville) [reply apis/outlook/mail/code/pkg/printers/mail.go]: fixed

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (g-linville) [reply apis/outlook/mail/code/pkg/printers/mail.go]: fixed

File: apis/outlook/mail/code/pkg/commands/searchMessages.go
Comment (g-linville) [reply apis/outlook/mail/code/pkg/commands/searchMessages.go]: fixed

File: apis/outlook/mail/code/go.mod
Comment (g-linville) [reply apis/outlook/mail/code/go.mod:5]: I had separate Go modules for the manage and read tools, and want this code to be usable by both of them without one of them being the distinct "owner" of it

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (thedadams) [inline apis/outlook/mail/code/pkg/printers/mail.go:89]: Daishan's comment below caused me to look into this: `models.MailFolderable` is an interface, so you're passing a pointer to an interface (which is why you have to dereference to use it).

Since you are "addressing" the variable yourself when you pass to these "replace IDs" functions, I would suggest not doing that and passing the variable itself (which is likely already a pointer because it's methods are mutating).

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (thedadams) [inline apis/outlook/mail/code/pkg/printers/mail.go:90]: Because of the way Go's type system works, after making the changes suggested above, this nil check doesn't really make sense. It won't protect you from a nil-pointer dereference.

You may already know that, but just in case: https://go.dev/play/p/FGVP-LqqzpT

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (g-linville) [reply apis/outlook/mail/code/pkg/printers/mail.go]: Fixed

File: apis/outlook/mail/code/pkg/printers/mail.go
Comment (g-linville) [reply apis/outlook/mail/code/pkg/printers/mail.go]: Fixed

**Review by thedadams — APPROVED** [review]

**Pull request merged**
